Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -362,6 +362,8 @@ static void clearFullscreenFlags(Element& element)
     element.setFullscreenFlag(false);
     if (auto* iframe = dynamicDowncast<HTMLIFrameElement>(element))
         iframe->setIFrameFullscreenFlag(false);
+
+    element.document().fullscreenManager().updatePageFullscreenStatusIfTopDocument();
```

Comment:
I don't think it's good that this function now has this side effect but its name says that it just clears a flag. I recommend thinking of a different name for the function.

Reply:
`topDocumentHasFullscreenElement()` is a flag too.

Maybe `updateFullscreenFlagsForExit` ? Though this doesn't seem particularly better than `clearFullscreenFlags` to me.

Reply:
Or `clearFullscreenFlagsAndUpdateTopDocumentStatus`? though that feels unnecessarily detailed.

You generally don't want to do clear the flags without also updating the top document flag, so I don't know if the distinction is super useful